nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/client/ESRestClientFactory.java

```diff
@@ -90,15 +98,27 @@ protected ESClient createLocalRestClient(ElasticSearchEmbeddedServerConfig serve
     }
 
     protected ESClient createRestClient(ElasticSearchClientConfig config) {
-        String[] hosts = config.getOption("addressList", "").split(",");
-        if (hosts.length == 0) {
+        String addressList = config.getOption("addressList", "");
+        if (addressList.isEmpty()) {
             throw new IllegalArgumentException("No addressList option provided cannot connect RestClient");
         }
+        String[] hosts = addressList.split(",");
         HttpHost[] httpHosts = new HttpHost[hosts.length];
         int i = 0;
         for (String host : hosts) {
-            String[] address = host.split(":");
-            httpHosts[i++] = new HttpHost(address[0], Integer.parseInt(address[1]));
+            try {
+                int hostPort = DEFAULT_HOST_PORT;
+                if (!host.startsWith("http")) {
+                    host = DEFAULT_SCHEMA_HTTP +host;
+                }
+                URL hostUrl = new URL(host);
+                if (hostUrl.getPort() != -1) {
+                    hostPort = hostUrl.getPort();
+                }
+                httpHosts[i++] = new HttpHost(hostUrl.getHost(), hostPort, hostUrl.getProtocol());
```

Comment:
Why not using `HttpHost#create(String)` ? 
It seems to handle schema and/or port presence with a fallback in both cases.

Reply:
Sound great.  I will change it.